package/MDAnalysis/coordinates/LAMMPS.py

```diff
@@ -443,3 +452,126 @@ def write(self, selection, frame=None):
 
             if has_velocities:
                 self._write_velocities(atoms)
+
+
+class DumpReader(base.ReaderBase):
+    """Reads the default `LAMMPS dump format`_
+
+    Expects trajectories produced by the default 'atom' style dump.
+
+    Will automatically convert positions from their scaled/fractional
+    representation to their real values.
+
+    .. versionadded:: 0.18.1
+    """
+    format = 'LAMMPSDUMP'
+
+    def __init__(self, filename, **kwargs):
+        super(DumpReader, self).__init__(filename, **kwargs)
+
+        root, ext = os.path.splitext(self.filename)
+        self._cache = {}
+
+        self._reopen()
+
+        self._read_next_timestep()
+
+    def _reopen(self):
+        try:
+            self._file.close()
+        except AttributeError:
+            pass
+        self._file = util.anyopen(self.filename)
+        self.ts = self._Timestep(self.n_atoms, **self._ts_kwargs)
+        self.ts.frame = -1
+
+    @property
+    @cached('n_atoms')
+    def n_atoms(self):
+        with util.anyopen(self.filename) as f:
+            f.readline()
+            f.readline()
+            f.readline()
+            n_atoms = int(f.readline())
+        return n_atoms
+
+    @property
+    @cached('n_frames')
+    def n_frames(self):
+        # 2(timestep) + 2(natoms info) + 4(box info) + 1(atom header) + n_atoms
+        lines_per_frame = self.n_atoms + 9
+        offsets = []
+        counter = 0
+        with util.anyopen(self.filename) as f:
+            line = True
+            while line:
+                if not counter % lines_per_frame:
+                    offsets.append(f.tell())
+                line = f.readline()
+                counter += 1
+        self._offsets = offsets[:-1]  # last is EOF
+        return len(self._offsets)
+
+    def close(self):
+        self._file.close()
+
+    def _read_frame(self, frame):
+        self._file.seek(self._offsets[frame])
```

Comment:
Does `seek` work on compressed files?

Reply:
Apparently yes, we test seeking and reading which relies on this